### DISCUSSION TITLE
Compatibility with CustomTKinter

### DISCUSSION THREAD

**Discussion by LunaAsInTheMoon** [insolor/async-tkinter-loop#49] [Ideas]
You may not have hear about it but there is a project called [Customtkinter](https://github.com/TomSchimansky/CustomTkinter) by @TomSchimansky. I am currently working on a project that uses this version of tkinter and your async handler would be a god send if this compatibility could be added.

**Comment by insolor** [comment]
@AbsolUnit, I've heard about it, but never tried it before.

Actually, it seems it works with minimal changes, at least on Linux:
```
import customtkinter
from async_tkinter_loop import async_handler, async_mainloop
import asyncio

customtkinter.set_appearance_mode("System")  # Modes: system (default), light, dark
customtkinter.set_default_color_theme("blue")  # Themes: blue (default), dark-blue, green

app = customtkinter.CTk()  # create CTk window like you do with the Tk window
app.geometry("400x240")


@async_handler
async def button_function():
    print("1")
    await asyncio.sleep(.5)
    print("2")


# Use CTkButton instead of tkinter Button
button = customtkinter.CTkButton(master=app, text="CTkButton", command=button_function)
button.place(relx=0.5, rely=0.5, anchor=customtkinter.CENTER)

async_mainloop(app)
```

![image](https://github.com/insolor/async-tkinter-loop/assets/2442833/dd02683b-b1ea-49a5-beb8-634ddc9c9969)


But, `mainloop` of `CTk` class has additional code for Windows, I need to figure out how to run it without running of `Tk.mainloop`

**Reply by LunaAsInTheMoon** [reply]
My project is on windows unfortunately, I'll have a look at your project and see if I can figure it out and I'll let you know

**Reply by LunaAsInTheMoon** [reply]
I managed to get it working by replacing tkinter.Tk() with customtkinter.CTk() and simply copying the code in the mainloop you were talking about over, its a bit hacky and not at all elegant but it will work for my purposes for now.

` 

    while True:
        # Process all pending events
        while root.tk.dooneevent(_tkinter.DONT_WAIT) > 0:
            if not root._window_exists:
                if sys.platform.startswith("win"):
                    root._windows_set_titlebar_color(root._get_appearance_mode())
                    if not root._withdraw_called_before_window_exists and not root._iconify_called_before_window_exists:
                        # print("window dont exists -> deiconify in mainloop")
                        root.deiconify()
                root._window_exists = True
`

**Reply by insolor** [reply]
@AbsolUnit, yes, something like that, but there should not be `while True`. Just add this code before running of `async_mainloop(app)`, without `while True`.

**Reply by insolor** [reply]
I've just published a [release](https://github.com/insolor/async-tkinter-loop/releases/tag/0.9.0) with an experimental CustomTkinter support. But it will work only with OOP style code (the main window should be a class inherited from `CTk` and `AsyncCTk`), see an example here: https://github.com/insolor/async-tkinter-loop/blob/main/examples/custom_tkinter.py
